tests/components/mobile_app/test_http_api.py

```diff
@@ -63,6 +65,40 @@ async def test_registration(
     )
 
 
+async def test_reregistration(
+    hass: HomeAssistant,
+    hass_client: ClientSessionGenerator,
+    hass_admin_user: MockUser,
+    create_registrations: tuple[dict[str, Any], dict[str, Any]],
+) -> None:
+    """Test that reregistrations overwrite the current entry."""
+    await async_setup_component(hass, DOMAIN, {DOMAIN: {}})
+
+    api_client = await hass_client()
+    config_entry = hass.config_entries.async_entries("mobile_app")[1]
```

Comment:
I'd first assert that there are two config entries for this domain to make this line more clear.

Reply:
```suggestion
    config_entry = hass.config_entries.async_entries(DOMAIN)[1]
```